citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/parse/iterators/CrossRefIterator.java

```diff
@@ -0,0 +1,279 @@
+import edu.asu.diging.crossref.service.CrossrefConfiguration;
+import edu.asu.diging.crossref.service.CrossrefWorksService;
+import edu.asu.diging.crossref.service.impl.CrossrefWorksServiceImpl;
+
+public class CrossRefIterator implements BibEntryIterator {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private JobInfo info;
+    private List<BibEntry> articles;
+    private BibEntry article;
+    private int currentIndex;
+
+    private boolean iteratorDone = false;
+    private Map<String, String> typeMap;
+
+    private CrossrefWorksService crossrefService;
+
+
+
+    public CrossRefIterator(JobInfo info) {
+        this.info = info;
+        currentIndex = 0;
+        init();
+    }
+
+    private void init() {
+        crossrefService = new CrossrefWorksServiceImpl(CrossrefConfiguration.getDefaultConfig());
+        typeMap = new HashMap<String, String>();
+        typeMap.put("journal-article", Publication.ARTICLE);
+        typeMap.put("book", Publication.BOOK);
+        typeMap.put("book-chapter", Publication.BOOK_CHAPTER); 
+        typeMap.put("monograph", Publication.MONOGRAPH);
+        typeMap.put("journal-issue", Publication.JOURNAL_ISSUE);
+        typeMap.put("reference-entry", Publication.REFERNCE_ENTRY);
+        typeMap.put("posted-content", Publication.POSTED_CONTENT);
+        typeMap.put("component", Publication.COMPONENT);
+        typeMap.put("edited-book", Publication.EDITED_BOOK);
+        typeMap.put("proceedings-article", Publication.PROCEEDINGS_ARTICLE);
+        typeMap.put("dissertation", Publication.DISSERTATION);
+        typeMap.put("book-section", Publication.BOOK_SECTION);
+        typeMap.put("report-component", Publication.REPORT_COMPONENT);
+        typeMap.put("report", Publication.REPORT);
+        typeMap.put("peer-review", Publication.PEER_REVIEW);
+        typeMap.put("book-track", Publication.BOOK_TRACK);
+        typeMap.put("book-part", Publication.BOOK_PART);
+        typeMap.put("other", Publication.OTHER);
+        typeMap.put("journal-volume", Publication.JORUNAL_VOLUME);
+        typeMap.put("book-set", Publication.BOOK_SET);
+        typeMap.put("journal", Publication.JOURNAL);
+        typeMap.put("proceedings-series", Publication.PROCEEDINGS_SERIES);
+        typeMap.put("report-series", Publication.REPORT_SERIES);
+        typeMap.put("proceedings", Publication.PROCEEDINGS);
+        typeMap.put("database", Publication.DATABASE);
+        typeMap.put("standard", Publication.STANDARD);
+        typeMap.put("reference-book", Publication.REFERENCE_BOOK);
+        typeMap.put("grant", Publication.GRANT);
+        typeMap.put("dataset", Publication.DATASET);
+        typeMap.put("book-series", Publication.BOOK_SERIES);
+        parseCrossRef();
+
+    }
+
+    private void parseCrossRef() {
+
+        articles = new ArrayList<>();
+        for (String doi : info.getDois()) {
+            try {
+                Item item = crossrefService.get(doi);
```

Comment:
The way it is written right now, this should be done in `next` (only one) and also paced to make sure we don't bombard crossref with requests. Even better would probably be to pass in a list of dois and make only one request but that might not be implemented either in our connector or in the crossref api.